docs/architecture.rst

```diff
@@ -15,12 +41,96 @@ DynamoDB for missed notifications as necessary.
 There will be many more Push servers to handle the connection node, while more
 Endpoint nodes can be handled as needed for notification throughput.
 
-Table Rotation
-==============
+.. _cryptography:
+
+Cryptography
+============
+
+The HTTP endpoint URL's generated by the connection nodes contain encrypted
+information, the :term:`UAID` and :term:`Subscription` to send the message to.
+This means that they both must have the same ``CRYPTO_KEY`` supplied to each.
+
+See :meth:`~autopush.settings.AutopushSettings.make_endpoint` for the endpoint
+URL generator.
+
+If you are only running Autopush locally, you can skip to :ref:`running` as
+later topics in this document apply only to developing or production scale
+deployments of Autopush.
+
+DynamoDB Tables
+===============
+
+Autopush uses a single router table and multiple messages tables, one for
+each month of the year. On startup, Autopush will create the router table and
+a message table for the prior month and the current month of the year.
+
+For more information on DynamoDB tables, see
+http://docs.aws.amazon.com/amazondynamodb/latest/gettingstartedguide/Welcome.html
+
+Router Table Schema
+-------------------
+
+The router table stores metadata for a given :term:`UAID` as well as which
+month table should be used for clients with a ``router_type`` of ``webpush``.
+
+For :term:`Bridging`, additional bridge-specific data may be stored in the
+router record for a :term:`UAID`.
+
+============   ==============
+uaid           **partition key** - :term:`UAID`
+router_type    :term:`Router Type`
+node_id        Hostname of the connection node the client is connected to.
+connected_at   Precise time (in milliseconds) the client connected to the node.
+last_connect   **global secondary index** - year-month-hour that the client has
+               last connected.
+curmonth       Message table name to use for storing :term:`WebPush` messages.
+============   ==============
+
+Autopush uses an optimistic deletion policy for ``node_id`` to avoid delete
+calls when not needed. If an endpoint node discovers during a delivery attempt that
+the ``node_id`` on record does not have the client connected, it will clear the
+``node_id`` record for that :term:`UAID` in the router table.
```

Comment:
I get this sentence, but it seems a bit awkward. Does

> During a delivery attempt, the endpoint will check the ``node_id`` for the corresponding ``uaid``. If the client is not connected, it will clear the ``node_id`` record for that ``uaid`` in the router table.

sound better?